test/runtests.jl

```diff
@@ -34,3 +35,67 @@ end
         @test hash == "dc0af40e8bc944a5d38c049b9a5b33e80b1e7a9621ac30fb440e5a2b6a4192d7"
     end
 end
+
+@testset "add_artifact!(_, _, ::GistUploadResult)" begin
+    # Dummy `gist` result object
+    gist = ArtifactUtils.GistUploadResult(
+        SHA1("65279f9c8a3dd1e2cb654fdedbe8cd58889ae1bc"),
+        "JuliaMono.tar.gz",
+        "<localpath>",
+        "https://github.com/cormullion/juliamono/releases/download/v0.007/JuliaMono.tar.gz",
+        "f1ab65231cda7981531398644a58fd5fde8f367b681e1b8e9c35d9b2aacfcb1c",
+        false,
+    )
+    mktempdir() do tempdir
+        artifact_file = joinpath(tempdir, "Artifacts.toml")
+        add_artifact!(artifact_file, "JuliaMono", gist)
+        artifacts = TOML.parsefile(artifact_file)
+        @test artifacts == Dict{String,Any}(
+            "JuliaMono" => Dict{String,Any}(
+                "git-tree-sha1" => "65279f9c8a3dd1e2cb654fdedbe8cd58889ae1bc",
+                "download" => Any[
+                    Dict{String,Any}(
+                        "sha256" => "f1ab65231cda7981531398644a58fd5fde8f367b681e1b8e9c35d9b2aacfcb1c",
+                        "url" => "https://github.com/cormullion/juliamono/releases/download/v0.007/JuliaMono.tar.gz",
+                    ),
+                ],
+            ),
+        )
+    end
+    str = sprint(show, "text/plain", gist)
+    @test occursin("upload_to_gist(", str)
+    @test occursin("; private = false)", str)
+    @test occursin("url =", str)
+end
+
+@testset "artifact_from_directory" begin
+    mktempdir() do tempdir
+        write(joinpath(tempdir, "file"), "hello")
+        @test artifact_from_directory(tempdir) ==
+              SHA1("538e83d637ab07ada6d841aa2454e0d5af4e52b3")
```

Comment:
This seems to be different on Windows, different line endings perhaps?

Reply:
see below https://github.com/simeonschaub/ArtifactUtils.jl/pull/8#discussion_r767210865